### ISSUE TITLE
Mudraptor family fighting other monster causes strange effects to severing limbs, and cause crashes with petraptors

### ISSUE THREAD

**Issue by PokkeTigrex** [Regalis11/Barotrauma#10213]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

When playing around with petraptors, when letting them fight other mudraptors, I found that they managed to sever the entire lower half of a mudraptor, without actually killing it, making them "levitate" in a sense. The same can happen to the petraptors as well.
![image](https://user-images.githubusercontent.com/85914375/199591521-02bee27a-90a9-4573-896e-bdac4b6c5bf6.png)
![image](https://user-images.githubusercontent.com/85914375/199591548-779b35bb-45ad-4ca3-96e5-affc13ac8b5d.png)

When I tried making them fight tiger threshers, something even more bizzare happens, with them basically decapitating the thresher upper head while keeping it alive, being reduced to a single living limb that bounces all over the place.

![image](https://user-images.githubusercontent.com/85914375/199591769-e8be9dc6-2a73-4ed0-a4e7-2d1bb0ed8e82.png)

I do not know if this is related to a larger issue with limbs and severing overall or only tied to creature vs creature, as I couldn't replicate this behavior using weapons like the boarding axe on mudraptors.

At some point in playing around with what petraptors can sever, I ended up crashing my game twice, once when letting them fight a bone thresher outside and once with them either fighting mudraptors or tiger threshers. I had trouble replicating it again but I assume it had something to do with the severing jank, or the fact that there is more than 2 petmudraptors existing at the same time, but don't currently have the time to experiment more with it so I will submit the two crash logs for now. 

Edit: I found a decently consistent way of producing the crash. I think it has something to do with either the petraptor attacking a monster at a certain spot or certain time, or vice versa. Crash logs generated from these crashes are basically the same as the ones I've posted already too. 
Also played around and found mudraptors in general have the weird quirk of severing parts but keeping them alive. Normal mudraptors though do not seem to be causing crashes when trying to replicate it with them, so its likely petraptors are the culprit of the crash themselves.

### Reproduction steps

Spawn up to two mudraptor_pet (or more), and let them fight a mudraptor or tiger thresher within the sub.

For the crash, spawn a fairly large amount of tiger threshers and petraptors in a ship ballast and let them fight. Berilla's ballasts works pretty well. I've had it crash in much smaller numbers, but it makes the crash occur much faster with more creatures attacks being thrown around at once to hit whatever crashes the game.

### Bug prevalence

Happens regularly

### Version

Other

### -

0.20.2.0 Unstable

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
Log #1
Barotrauma Client crash report (generated on 11/02/2022 15:25:13)


Barotrauma seems to have crashed. Sorry for the inconvenience! 


3CB2772E84E4404533525A516FC12A2F


Game version 0.20.2.0 (UnstableWindows, branch dev, revision 9abb4d2d84)
Graphics mode: 1920x1080 (BorderlessWindowed)
VSync OFF
Language: English
Selected content packages: Vanilla, Gato Revamped
Level seed: no level loaded
Loaded submarine: Azimuth (25AE622445B2A2D7075B90A31582C5AD)
Selected screen: Barotrauma.GameScreen
SteamManager initialized


System info:
    Operating system: Microsoft Windows NT 10.0.19045.0 64 bit
    GPU name: Intel(R) HD Graphics 530
    Display mode: {Width:1920 Height:1080 Format:Color AspectRatio:1.7777778}
    GPU status: Normal


Exception: Object reference not set to an instance of an object. (System.NullReferenceException)
Target site: Boolean UpdateLimbAttack(Single, Microsoft.Xna.Framework.Vector2, Barotrauma.IDamageable, Single, Barotrauma.Limb)
Stack trace: 
   at Barotrauma.EnemyAIController.UpdateLimbAttack(Single deltaTime, Vector2 attackSimPos, IDamageable damageTarget, Single distance, Limb targetLimb) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\EnemyAIController.cs:line 2141
   at Barotrauma.EnemyAIController.UpdateAttack(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\EnemyAIController.cs:line 1901
   at Barotrauma.EnemyAIController.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\EnemyAIController.cs:line 441
   at Barotrauma.Character.UpdateAll(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 2920
   at Barotrauma.GameScreen.Update(Double deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Screens\GameScreen.cs:line 192
   at Barotrauma.GameMain.Update(GameTime gameTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 860
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 656
   at Microsoft.Xna.Framework.Game.Tick() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 500
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 92
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 397
   at Microsoft.Xna.Framework.Game.Run() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 367
   at Barotrauma.Program.Main(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58


Last debug messages:
[11/02/2022 15:25:12] spawn Tigerthresher cursor
[11/02/2022 15:25:10] spawn Tigerthresher cursor'
[11/02/2022 15:24:54] spawn Mudraptor_pet cursor
[11/02/2022 15:24:53] spawn Mudraptor_pet cursor
[11/02/2022 15:24:53] spawn Mudraptor_pet cursor
[11/02/2022 15:24:53] spawn Mudraptor_pet cursor
[11/02/2022 15:24:53] spawn Mudraptor_pet cursor
[11/02/2022 15:24:52] spawn Mudraptor_pet cursor
[11/02/2022 15:24:52] spawn Mudraptor_pet cursor
[11/02/2022 15:24:52] spawn Mudraptor_pet cursor
[11/02/2022 15:24:41] spawn Tigerthresher cursor
[11/02/2022 15:24:41] spawn Tigerthresher cursor
[11/02/2022 15:24:41] spawn Tigerthresher cursor
[11/02/2022 15:24:27] spawn Tigerthresher cursor
[11/02/2022 15:24:25] spawn Mudraptor_pet cursor
[11/02/2022 15:24:25] spawn Mudraptor_pet cursor
[11/02/2022 15:24:25] spawn Mudraptor_pet cursor
[11/02/2022 15:24:18] freecam
[11/02/2022 15:24:01] spawn Mudraptor_pet cursor
[11/02/2022 15:24:01] spawn Mudraptor_pet cursor
[11/02/2022 15:24:00] spawn Mudraptor_pet cursor
[11/02/2022 15:23:59] spawn Mudraptor_pet cursor
[11/02/2022 15:23:55] spawn Tigerthresher cursor
[11/02/2022 15:23:38] spawn Tigerthresher cursor
[11/02/2022 15:23:30] spawn Mudraptor_pet cursor
[11/02/2022 15:23:29] spawn Mudraptor_pet cursor
[11/02/2022 15:23:28] spawn Mudraptor_pet cursor
[11/02/2022 15:23:21] fixhulls
[11/02/2022 15:22:56] despawnnow
[11/02/2022 15:22:26] spawn Defensebot cursor
[11/02/2022 15:22:25] spawn Defensebot cursor
[11/02/2022 15:22:00] spawn Mudraptor_pet cursor
[11/02/2022 15:22:00] spawn Mudraptor_pet cursor
[11/02/2022 15:21:59] spawn Mudraptor_pet cursor
[11/02/2022 15:21:59] spawn Mudraptor_pet cursor
[11/02/2022 15:21:55] spawn Mudraptor_pet
[11/02/2022 15:21:36] spawn Mudraptor_pet cursor
[11/02/2022 15:21:35] spawn Mudraptor_pet cursor
[11/02/2022 15:21:35] spawn Mudraptor_pet cursor
[11/02/2022 15:21:35] spawn Mudraptor_pet cursor
[11/02/2022 15:21:16] spawn Mudraptor_pet cursor
[11/02/2022 15:21:16] spawn Mudraptor_pet cursor
[11/02/2022 15:21:15] spawn Mudraptor_pet cursor
[11/02/2022 15:20:07] despawnnow
[11/02/2022 15:20:00] spawn Mudraptor_pet cursor
[11/02/2022 15:19:58] spawn Mudraptor_pet cursor
[11/02/2022 15:19:42] spawn Mudraptor_pet cursor
[11/02/2022 15:19:42] spawn Mudraptor_pet cursor
[11/02/2022 15:19:26] spawn Mudraptor_pet cursor
[11/02/2022 15:19:26] spawn Mudraptor_pet cursor
[11/02/2022 15:17:25] despawnnow
[11/02/2022 15:16:57] spawn Mudraptor cursor
[11/02/2022 15:16:57] spawn Mudraptor cursor
[11/02/2022 15:16:56] spawn Mudraptor cursor
[11/02/2022 15:16:01] spawn Mudraptor cursor
[11/02/2022 15:15:09] spawn Mudraptor cursor
[11/02/2022 15:15:05] fixhulls
[11/02/2022 15:14:13] spawn Crawler cursor
[11/02/2022 15:14:12] spawn Crawler cursor
[11/02/2022 15:14:12] spawn Crawler cursor
[11/02/2022 15:14:11] spawn Crawler cursor
[11/02/2022 15:14:10] spawn Crawler cursor
[11/02/2022 15:13:57] spawn Crawler cursor
[11/02/2022 15:13:14] spawn Crawler cursor
[11/02/2022 15:13:14] spawn Crawler cursor
[11/02/2022 15:13:13] spawn Crawler cursor
[11/02/2022 15:12:56] spawn Crawler cursor
[11/02/2022 15:11:46] fixitems
[11/02/2022 15:10:44] Gave talent "Fireman's Carry" to "Althea Dimauro".
[11/02/2022 15:10:44] givetalent "Fireman's Carry"
[11/02/2022 15:10:03] spawn Defensebot cursor
[11/02/2022 15:09:55] spawn Defensebot cursor
[11/02/2022 15:09:21] spawn Mudraptor_pet cursor
[11/02/2022 15:08:36] spawnitem "Petraptor egg" inventory
[11/02/2022 15:08:17] spawnitem Saline inventory 8
[11/02/2022 15:08:12] spawnitem Saline inventory
[11/02/2022 15:08:03] spawnitem "Petraptor egg" inventory
[11/02/2022 15:06:09] spawn assistant cursor
[11/02/2022 15:05:11] spawn assistant cursor
[11/02/2022 15:04:56] spawn assistant cursor
[11/02/2022 15:04:25] spawn assistant cursor
[11/02/2022 15:04:20] spawnitem "Hardened Crowbar" inventory
[11/02/2022 15:04:13] Gave talent "Ph.D in Nuclear Physics" to "Micheal Kinney".
[11/02/2022 15:04:13] givetalent "Ph.D in Nuclear Physics"
[11/02/2022 15:03:18] fixitems
[11/02/2022 15:02:34] spawn assistant cursor
[11/02/2022 15:02:20] spawn assistant cursor
[11/02/2022 15:01:55] spawn assistant cursor
[11/02/2022 15:01:46] spawnitem Screwdriver inventory 2
[11/02/2022 15:01:19] spawn assistant cursor
[11/02/2022 15:00:59] spawn assistant cursor
[11/02/2022 15:00:54] spawnitem "Hardened Screwdriver" inventory 2
[11/02/2022 15:00:48] Gave talent "Aggressive Engineering" to "Micheal Kinney".
[11/02/2022 15:00:48] givetalent "Aggressive Engineering" 
[11/02/2022 15:00:26] spawn engineer cursor
[11/02/2022 14:59:52] spawn assistant cursor
[11/02/2022 14:58:59] spawn assistant cursor
[11/02/2022 14:58:26] Gave Carlos Phoenix 20000 experience
[11/02/2022 14:58:26] giveexperience 20000
[11/02/2022 14:58:21] spawnitem "Heavy Wrench" inventory
[11/02/2022 14:57:48] spawn mechanic cursor
[11/02/2022 14:57:41] kill
[11/02/2022 14:57:11] spawn assistant cursor
[11/02/2022 14:56:54] spawn assistant cursor
[11/02/2022 14:56:25] spawn assistant cursor
[11/02/2022 14:55:56] spawn assistant cursor
[11/02/2022 14:55:28] spawn assistant cursor
[11/02/2022 14:55:04] spawn assistant cursor
[11/02/2022 14:54:46] spawn assistant cursor
[11/02/2022 14:54:21] spawn assistant cursor
[11/02/2022 14:54:08] spawn assistant cursor
[11/02/2022 14:53:51] spawn assistant cursor
[11/02/2022 14:53:29] spawn assistant cursor
[11/02/2022 14:53:23] spawnitem Iron inventory 36
[11/02/2022 14:52:40] spawn assistant cursor
[11/02/2022 14:52:04] spawn assistant cursor
[11/02/2022 14:49:50] revive "Austin Price"
[11/02/2022 14:49:34] Set Winston McLeroy's weapons level to 100
[11/02/2022 14:49:34] setskill weapons 100
[11/02/2022 14:49:06] spawn assistant cursor
[11/02/2022 14:48:30] Gave talent "Inordinate Exsanguination" to "Winston McLeroy".
[11/02/2022 14:48:30] givetalent "Inordinate Exsanguination"
[11/02/2022 14:48:17] spawn securityofficer cursor
[11/02/2022 14:47:01] spawn assistant cursor
[11/02/2022 14:46:04] spawnitem "Battery Cell" inventory 6
[11/02/2022 14:44:41] spawnitem Lead inventory
[11/02/2022 14:44:40] spawnitem Lead inventory
[11/02/2022 14:44:36] spawnitem "FPGA Circuit" inventory
[11/02/2022 14:44:32] spawnitem "Titanium-Aluminum Alloy" inventory 3
[11/02/2022 14:44:22] Set Winston McLeroy's mechanical level to 100
[11/02/2022 14:44:22] setskill mechanical 100
[11/02/2022 14:43:53] spawnitem "Metal Crate" inventory
[11/02/2022 14:43:42] spawnitem Scrap inventory 24
[11/02/2022 14:42:35] Gave Winston McLeroy 20000 experience
[11/02/2022 14:42:35] giveexperience 20000
[11/02/2022 14:42:02] Lighting disabled
[11/02/2022 14:42:02] lighting
[11/02/2022 14:41:53] spawnitem "Oxygenite Tank" inventory
[11/02/2022 14:41:49] spawnitem "Battery Cell" inventory
[11/02/2022 14:41:36] spawnitem Exosuit inventory
[11/02/2022 14:39:31] Attempting to open ALC device "OpenAL Soft on Speakers (Logitech Wireless Headset)"
[11/02/2022 14:39:28] Logged in as Tigrex (SteamID STEAM_1:0:33529024)

Log #2
Barotrauma Client crash report (generated on 11/02/2022 15:28:49)


Barotrauma seems to have crashed. Sorry for the inconvenience! 


3CB2772E84E4404533525A516FC12A2F


Game version 0.20.2.0 (UnstableWindows, branch dev, revision 9abb4d2d84)
Graphics mode: 1920x1080 (BorderlessWindowed)
VSync OFF
Language: English
Selected content packages: Vanilla, Gato Revamped
Level seed: no level loaded
Loaded submarine: Herja (FF29A53003B8C4DC2CFDBA1BD72BCF3F)
Selected screen: Barotrauma.GameScreen
SteamManager initialized


System info:
    Operating system: Microsoft Windows NT 10.0.19045.0 64 bit
    GPU name: Intel(R) HD Graphics 530
    Display mode: {Width:1920 Height:1080 Format:Color AspectRatio:1.7777778}
    GPU status: Normal


Exception: Object reference not set to an instance of an object. (System.NullReferenceException)
Target site: Boolean UpdateLimbAttack(Single, Microsoft.Xna.Framework.Vector2, Barotrauma.IDamageable, Single, Barotrauma.Limb)
Stack trace: 
   at Barotrauma.EnemyAIController.UpdateLimbAttack(Single deltaTime, Vector2 attackSimPos, IDamageable damageTarget, Single distance, Limb targetLimb) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\EnemyAIController.cs:line 2204
   at Barotrauma.EnemyAIController.UpdateAttack(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\EnemyAIController.cs:line 1901
   at Barotrauma.EnemyAIController.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\EnemyAIController.cs:line 441
   at Barotrauma.Character.UpdateAll(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 2920
   at Barotrauma.GameScreen.Update(Double deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Screens\GameScreen.cs:line 192
   at Barotrauma.GameMain.Update(GameTime gameTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 860
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 656
   at Microsoft.Xna.Framework.Game.Tick() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 500
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 92
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 397
   at Microsoft.Xna.Framework.Game.Run() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 367
   at Barotrauma.Program.Main(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58


Last debug messages:
[11/02/2022 15:28:43] fixhulls
[11/02/2022 15:28:40] spawn Mudraptor_pet cursor
[11/02/2022 15:28:39] spawn Mudraptor_pet cursor
[11/02/2022 15:28:39] spawn Mudraptor_pet cursor
[11/02/2022 15:28:38] spawn Mudraptor_pet cursor
[11/02/2022 15:28:37] spawn Mudraptor_pet cursor
[11/02/2022 15:28:35] spawn Mudraptor_pet cursor
[11/02/2022 15:28:34] spawn Mudraptor_pet cursor
[11/02/2022 15:28:34] spawn Mudraptor_pet cursor
[11/02/2022 15:28:29] spawn Bonethresher cursor
[11/02/2022 15:27:36] despawnnow
[11/02/2022 15:27:25] kill Mudraptor_pet 0
[11/02/2022 15:27:22] kill Mudraptor_pet 1
[11/02/2022 15:27:21] Found multiple matching characters. Use "[charactername] [0-1]" to choose a specific character.
[11/02/2022 15:27:21] kill Mudraptor_pet
[11/02/2022 15:27:14] spawn Tigerthresher cursor
[11/02/2022 15:27:12] spawn Mudraptor_pet cursor
[11/02/2022 15:27:10] spawn Mudraptor_pet cursor
[11/02/2022 15:27:02] freecamera
[11/02/2022 15:26:59] Command "freeca," not found.
[11/02/2022 15:26:59] freeca,
[11/02/2022 15:26:42] Lighting disabled
[11/02/2022 15:26:42] lighting
[11/02/2022 15:26:40] freecamera
[11/02/2022 15:25:31] Attempting to open ALC device "OpenAL Soft on Speakers (Logitech Wireless Headset)"
[11/02/2022 15:25:29] Logged in as Tigrex (SteamID STEAM_1:0:33529024)
```

**Comment by PokkeTigrex** [comment]
I managed to replicate the crash I've been having, I updated my issue with the steps.

**Comment by itchyOwl** [comment]
Caused by Regalis11/Barotrauma-development@684a160, fixed in Regalis11/Barotrauma-development@582f78d.

**Comment by Rokvach** [comment]
Tested, working correctly. Closing.
